cleanlab/count.py

```diff
@@ -1108,12 +1111,31 @@ def get_confident_thresholds(labels: np.array, pred_probs: np.array) -> np.array
       class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
       higher) fold cross-validation.
 
+    multi_label : bool, optional
+      If ``True``, labels should be an iterable (e.g. list) of iterables, containing a
+      list of labels for each example, instead of just a single label.
+      The multi-label setting supports classification tasks where an example has 1 or more labels.
+      Example of a multi-labeled `labels` input: ``[[0,1], [1], [0,2], [0,1,2], [0], [1], ...]``.
+      The major difference in how this is calibrated versus single-label is that
+      the total number of errors considered is based on the number of labels,
+      not the number of examples. So, the calibrated `confident_joint` will sum
+      to the number of total labels.
+
     Returns
     -------
     confident_thresholds : np.array
       An array of shape ``(K,)``.
     """
-    confident_thresholds = np.array(
-        [np.mean(pred_probs[:, k][labels == k]) for k in range(pred_probs.shape[1])]
-    )
+
+    if multi_label:
+        # Compute unique number of classes K by flattening labels (list of lists)
+        K = len(np.unique([i for lst in labels for i in lst]))
+        # Compute thresholds = p(label=k | k in set of given labels)
+        k_in_l = np.array([[k in lst for lst in labels] for k in range(K)])
+        # The avg probability of class given that the label is represented.
+        confident_thresholds = [np.mean(pred_probs[:, k][k_in_l[k]]) for k in range(K)]
+    else:
+        confident_thresholds = np.array(
+            [np.mean(pred_probs[:, k][labels == k]) for k in range(pred_probs.shape[1])]
+        )
     return confident_thresholds
```

Comment:
seems like `confident_thresholds` is not always a np.array. I'd remove the `np.array(...)` wrapper from L1138 and instead put it at the return in L1141:

```
return np.array(confident_thresholds)
```

Reply:
Isn't the return type of this a list for the multilabel setting?  

Current docstring for the function says it returns np.array always